File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int>set1(nums1.begin() , nums1.end());
        unordered_set<int>intersection_set;
        for(int num : nums2){
            if(set1.count(num)){
                intersection_set.insert(num);
            }
        }
        return vector<int>(intersection_set.begin(),intersection_set.end());
    }
};

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer>set1 = new HashSet<>();
        for(int num:nums1){
            set1.add(num);
        }
        List<Integer>intersectionList = new ArrayList<>();
        for(int num:nums2){
            if(set1.contains(num)){
                intersectionList.add(num);
                set1.remove(num);
            }
        }
        int[] intersectionArray = new int[intersectionList.size()];
        for(int i=0;i<intersectionList.size();i++){
            intersectionArray[i]=intersectionList.get(i);
            
        }
        return intersectionArray;
    }
}


